### PR TITLE
Add tag support by name convention

### PR DESCRIPTION
**Pull request by cliscum**
To tag a metric, name it like my.metric[tag1:foo,tag2:bar,...]

### CODE REVIEW

File: metrics_reporter_test.go [hunk 1]
```diff
@@ -155,3 +155,21 @@ func (_ *ReporterSuite) TestTimerSeries(c *C) {
...
+	counter.Inc(222)
+	series := reporter.series(
+		t.Unix(), "my.counter[food:bagels,drink:coffee]", counter)
+	c.Check(series, HasLen, 1)
+	s := series[0]
+	c.Check(s.Metric, Equals, "my.counter.count")
+	c.Check(s.Type, Equals, "counter")
+	c.Check(s.Points, HasLen, 1)
+	c.Check(s.Points[0][0], Equals, t.Unix())
+	c.Check(s.Points[0][1], Equals, int64(666))
```
Comment (cliscum) [inline metrics_reporter_test.go]: I'm going to get rid of lines 168-171 here. Not intending to test that stuff, and it's covered elsewhere.

**Comment by markchadwick** [comment]
:+1:

**Comment by jawsthegame** [comment]
:+1: :+1:

**Comment by chrismcbride** [comment]
:100:

**Pull request merged**